data-protocols/dsp/dsp-api-configuration/src/test/java/org/eclipse/edc/protocol/dsp/api/configuration/DspApiConfigurationExtensionTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ *  Copyright (c) 2023 Fraunhofer Institute for Software and Systems Engineering
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Fraunhofer Institute for Software and Systems Engineering - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.protocol.dsp.api.configuration;
+
+import org.eclipse.edc.junit.extensions.DependencyInjectionExtension;
+import org.eclipse.edc.spi.iam.IdentityService;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+import org.eclipse.edc.spi.system.configuration.ConfigFactory;
+import org.eclipse.edc.spi.system.injection.ObjectFactory;
+import org.eclipse.edc.spi.types.TypeManager;
+import org.eclipse.edc.web.spi.WebServer;
+import org.eclipse.edc.web.spi.WebService;
+import org.eclipse.edc.web.spi.configuration.WebServiceConfiguration;
+import org.eclipse.edc.web.spi.configuration.WebServiceConfigurer;
+import org.eclipse.edc.web.spi.configuration.WebServiceConfigurerImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.edc.protocol.dsp.api.configuration.DspApiConfigurationExtension.CONTEXT_ALIAS;
+import static org.eclipse.edc.protocol.dsp.api.configuration.DspApiConfigurationExtension.DEFAULT_DSP_WEBHOOK_ADDRESS;
+import static org.eclipse.edc.protocol.dsp.api.configuration.DspApiConfigurationExtension.DSP_WEBHOOK_ADDRESS;
+import static org.eclipse.edc.protocol.dsp.api.configuration.DspApiConfigurationExtension.SETTINGS;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(DependencyInjectionExtension.class)
```

Comment:
Can we convert this to unit tests?

Reply:
When converting this to a unit test, the injected services would have to be set on the extension by e.g. using reflection, not sure we want that. Or is there another way to construct the extension class that I'm missing?

Reply:
You could use `@ExtendWith(DependencyInjectionExtension.class)`. 

However, my opinion is we should be injecting on protected fields, not private, as that is common practice in Java (which would solve this issue). I realize private field injection is used throughout the codebase but I would be in favor of changing that.

Reply:
If it's something that would need to be changed throughout the code base, there should be a separate PR anyway. I would then leave the test as is for the time being, as it is set up in the same way as other api configuration tests (ids, management). Refactoring could then be done on all of those at once. Wdyt?

Reply:
I think there are two issues here:

1. Refactor the codebase to cleanup injection. We would need to discuss this, and I agree this is best done later after this release.

2. Clean this test up and break it into targetted unit tests. This could be done using actors or the dependency injection extension.

I think we should do #2 as I am currently trying to get rid of a lot of these "unit tests masquerading as integration tests" things.